src/main/java/net/mcreator/ui/workspace/WorkspacePanel.java

```diff
@@ -1218,50 +1218,28 @@ public void editCurrentlySelectedModElement(ModElement mu, JComponent component,
 	}
 
 	private void editCurrentlySelectedModElementAsCode(ModElement mu, JComponent component, int x, int y) {
-		GeneratableElement ge = mu.getGeneratableElement();
-
-		List<GeneratorTemplate> modElementFiles = mcreator.getGenerator().getModElementGeneratorTemplatesList(mu, ge);
-
-		modElementFiles.addAll(
-				mcreator.getGenerator().getModElementGlobalTemplatesList(mu.getType(), false, new AtomicInteger()));
-
-		if (ge instanceof ICommonType) {
-			Collection<BaseType> baseTypes = ((ICommonType) ge).getBaseTypesProvided();
+		List<GeneratorTemplate> modElementFiles = mcreator.getGenerator().getModElementGeneratorTemplatesList(mu)
+				.stream().filter(e -> !e.isListTemplate()).toList();
+		List<GeneratorTemplate> modElementGlobalFiles = mcreator.getGenerator()
+				.getModElementGlobalTemplatesList(mu.getType(), false, new AtomicInteger());
+		List<GeneratorTemplatesList> modElementListFiles = mcreator.getGenerator().getModElementListTemplates(mu);
+
+		if (mu.getGeneratableElement() instanceof ICommonType) {
+			Collection<BaseType> baseTypes = ((ICommonType) mu.getGeneratableElement()).getBaseTypesProvided();
 			for (BaseType baseType : baseTypes) {
-				modElementFiles.addAll(mcreator.getGenerator().getGlobalTemplatesList(
+				modElementGlobalFiles.addAll(mcreator.getGenerator().getGlobalTemplatesList(
 						mcreator.getGenerator().getGeneratorConfiguration().getDefinitionsProvider()
 								.getBaseTypeDefinition(baseType), false, new AtomicInteger()));
 			}
 		}
 
-		if (modElementFiles.size() > 1) {
-			JPopupMenu codeDropdown = new JPopupMenu();
-			codeDropdown.setBorder(BorderFactory.createEmptyBorder());
-			codeDropdown.setBackground(((Color) UIManager.get("MCreatorLAF.LIGHT_ACCENT")).darker());
-
-			boolean global = false;
-			for (GeneratorTemplate modElementFile : modElementFiles) {
-				if (!global && modElementFile.isGlobal()) {
-					codeDropdown.addSeparator();
-					global = true;
-				}
-
-				JMenuItem item = new JMenuItem(
-						"<html>" + modElementFile.getFile().getName() + "<br><small color=#666666>"
-								+ mcreator.getWorkspace().getWorkspaceFolder().toPath()
-								.relativize(modElementFile.getFile().toPath()));
-				item.setIcon(FileIcons.getIconForFile(modElementFile.getFile()));
-				item.setBackground(((Color) UIManager.get("MCreatorLAF.LIGHT_ACCENT")).darker());
-				item.setForeground((Color) UIManager.get("MCreatorLAF.BRIGHT_COLOR"));
-				item.setIconTextGap(8);
-				item.setBorder(BorderFactory.createEmptyBorder(3, 0, 5, 3));
-				item.addActionListener(e -> ProjectFileOpener.openCodeFile(mcreator, modElementFile.getFile()));
-				codeDropdown.add(item);
-			}
-			codeDropdown.show(component, x, y);
-		} else if (modElementFiles.size() == 1) {
+		if (modElementFiles.size() + modElementGlobalFiles.size() > 1 || modElementListFiles.size() > 0)

```

Comment:
This will now still open dropdown if there is only one modElementListFile

Reply:
List files are not usual mod element files, their groups are like sub-categories of ME files. Showing menu in this case indicates that generated file does not belong to the default category (which is not shown if `modElementFiles.size() + modElementGlobalFiles.size() > 1`).

Reply:
Why does the user need this indication if there is only one file?

If ME does not generate any files but list files and only one is present, not sure if they need to know this?

Reply:
OKay, I'll let it be...

Reply:
✔️ Fixed